### PR TITLE
Add dependencies to the `Protobuf` external project

### LINKED ISSUE #141: build error when `LINK_STATIC_LIBPROTOBUF=ON` and `make -j` are used
```
$ cmake -DLINK_STATIC_LIBPROTOBUF=ON .. && make -j4
...
Scanning dependencies of target gen_onnx_proto
Scanning dependencies of target Protobuf
[  1%] Running gen_proto.py on onnx/onnx.in.proto
[  2%] Creating directories for 'Protobuf'
[  3%] Performing download step (download, verify and extract) for 'Protobuf'
-- Downloading...
   dst='/Users/sakai/Menoh/build/downloads/protobuf-cpp-3.6.1.tar.gz'
   timeout='none'
-- Using src='https://github.com/protocolbuffers/protobuf/releases/download/v3.6.1/protobuf-cpp-3.6.1.tar.gz'
Processing /Users/sakai/Menoh/external/onnx/onnx/onnx.in.proto
Writing /Users/sakai/Menoh/build/external/onnx/onnx/onnx.proto
Writing /Users/sakai/Menoh/build/external/onnx/onnx/onnx.proto3
generating /Users/sakai/Menoh/build/external/onnx/onnx/onnx_pb.py
[  4%] Running C++ protocol buffer compiler on /Users/sakai/Menoh/build/external/onnx/onnx/onnx.proto
/bin/sh: ../../protobuf-3.6.1/bin/protoc: No such file or directory
make[2]: *** [external/onnx/onnx/onnx.pb.cc] Error 127
make[1]: *** [external/onnx/CMakeFiles/gen_onnx_proto.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
-- [download 0% complete]
...
-- [download 100% complete]
-- verifying file...
       file='/Users/sakai/Menoh/build/downloads/protobuf-cpp-3.6.1.tar.gz'
-- Downloading... done
-- extracting...
     src='/Users/sakai/Menoh/build/downloads/protobuf-cpp-3.6.1.tar.gz'
     dst='/Users/sakai/Menoh/build/protobuf-3.6.1/src/Protobuf'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[  6%] No patch step for 'Protobuf'
[  7%] No update step for 'Protobuf'
[  8%] Performing configure step for 'Protobuf'
...
```

[full log](https://github.com/pfnet-research/menoh/files/2476508/log.txt)

### PR DESCRIPTION
**Pull request by okapies**
Add explicit dependencies to the `Protobuf` external project to prevent running a command which uses `protoc` before building it especially in case of `make`ing targets in parallel (e.g. `-j4`).

Fixes  #141.

### CODE REVIEW

**Comment by okapies** [comment]
@msakai Could you check it in your environment? Thanks.

**Comment by msakai** [comment]
Thanks a lot!
It works on my macOS environment, though I don't quite understand why this works. :sweat_smile:

https://github.com/onnx/onnx/blob/bae6333e149a59a3faa9c4d9c44974373dcf5256/CMakeLists.txt#L218-L238

The custom command depends on `protobuf::protoc` target and adding `Protobuf` to its dependency forces the `Protobuf` target to be built before the custom command?

File: scripts/build-menoh.sh [hunk 1]
```diff
@@ -63,8 +63,8 @@ test -n "${ARG_BUILD_DIR}" || readonly ARG_BUILD_DIR="${ARG_SOURCE_DIR}/build"
 test -n "${ARG_INSTALL_DIR}" || readonly ARG_INSTALL_DIR=/usr/local
 
 if [ -n "${ARG_MKLDNN_DIR}" ]; then
-    OPT_MKLDNN_INCLUDE_DIR=-DMKLDNN_INCLUDE_DIR=${ARG_MKLDNN_DIR}/include
-    OPT_MKLDNN_LIBRARY=-DMKLDNN_LIBRARY=${ARG_MKLDNN_DIR}/lib/libmkldnn.so
+    readonly OPT_MKLDNN_INCLUDE_DIR=-DMKLDNN_INCLUDE_DIR=${ARG_MKLDNN_DIR}/include
+    readonly OPT_MKLDNN_LIBRARY=-DMKLDNN_LIBRARY=${ARG_MKLDNN_DIR}/lib/libmkldnn.so
```
Comment (msakai) [inline scripts/build-menoh.sh]: Is this change related to this issue?
If not, how about applying this change in separate PR?

**Comment by okapies** [comment]
Yes, because ONNX’s generator target depends on `protobuf::protoc`.

File: scripts/build-menoh.sh (hunk 1 above)
Comment (okapies) [reply scripts/build-menoh.sh]: It's just a fool-proof to avoid overriding variables from outside which is also done in [other place](https://github.com/pfnet-research/menoh/blob/38f65a8433e11edef039ecba47404264ed96936d/scripts/build-menoh.sh#L60-L63). I'd like to do it in this PR.

**Comment by msakai** [comment]
I see. Thanks for your explanation.

File: scripts/build-menoh.sh (hunk 1 above)
Comment (msakai) [reply scripts/build-menoh.sh]: Okay.

**Comment by okapies** [comment]
Thanks!
